libres/lib/analysis/enkf_linalg.cpp

```diff
@@ -35,10 +35,28 @@ void enkf_linalg_genX3(matrix_type *X3, const matrix_type *W,
     matrix_free(X2);
 }
 
+/*
+Implements a version of Eq. 14.34 from the book Data Assimilation,
+The Ensemble Kalman Filter, 2nd Edition by Geir Evensen.
+
+Eq. 14.34 is as follows:
+
+    X2 = (I + \Lambda_1)^{-1/2} @ X^T_1 * S
+
+, but this function seems to implement the following:
+
+    X2 = X^T_1 * S
+    X2 = (\Lambda^{-1/2}_1 * X2^T)^T
+
+See tests for more details.
+
+TODO: The differences between equation and implementation need explaining.
```

Comment:
Is this TODO going to be DONE in this PR? Suggest removing this line and making a separate issue on Github.

Reply:
Fixed, thanks.

Reply:
Removed and created this issue: https://github.com/equinor/ert/issues/2360

Thanks